File: other/training/triangle/answer/Myon.php
<?php
function input()
{
    return rtrim(fgets(STDIN));
}

function input_array()
{
    return explode(' ', input());
}

$num = input_array();
sort($num);
if ($num[0] + $num[1] > $num[2] || $num[1] + $num[2] > $num[3]) echo "TRIANGLE\n";
else if($num[0] + $num[1] == $num[2] || $num[1] + $num[2] == $num[3]) echo "SEGMENT\n";
else echo "IMPOSSIBLE\n";
?>

File: other/training/triangle/answer/Myon.cpp
#include <iostream>
#include <algorithm>
using namespace std;

int main()
{
    int num[4];
    cin >> num[0] >> num[1] >> num[2] >> num[3];
    sort(num, num + 4);
    if (num[0] + num[1] > num[2] || num[1] + num[2] > num[3]) cout << "TRIANGLE" << endl;
    else if(num[0] + num[1] == num[2] || num[1] + num[2] == num[3]) cout << "SEGMENT" << endl;
    else cout << "IMPOSSIBLE" << endl;
    return 0;
}
